### DISCUSSION TITLE
thread_cache_size의 default는 몇일까요?

### DISCUSSION THREAD

**Discussion by CODe5753** [Well-Founded-Confidence/RealMySQL#2] [Q&A]
default 사이즈가 몇일까요?
그리고 왜 default 사이즈가 그렇게 설정되어 있을까요?

**Comment by menthamin** [comment]
디폴트 사이즈는 아닌 것 같지만 저희 운영 환경 thread_cache_size를 확인해봤습니다.

thread_cache는 사용자(스레드)별로 할당되는 메모리 영역으로 알고 있어
너무 큰 size로 할당하면 MySQL 글로벌 영역에 할당할 메모리가 부족할 것 같습니다.

하지만 스레드 cache_size를 너무 작게 준다면 클라이언트가 쿼리 실행 시 Out of memory와 같은 에러가 발생할 수 있을 것 같아
적절한 값이 필요해 보입니다.

```
show variables where Variable_name like 'thread%';
```

<img width="861" alt="image" src="https://user-images.githubusercontent.com/29594498/229801459-8340a54f-76fa-4483-81db-530e90952805.png">

**Reply by CODe5753** [reply]
thread_cache_size는 `스레드의 수`를 제어하는 시스템 변수로 알고 있습니다.
즉, 이 size가 작은 경우엔 스레드 생성과 제거에 오버헤드가 자주 발생하게 될 것이고,
크다면 스레드의 총 갯수가 늘어나므로 메모리 사용량이 증가할 것 같습니다.
따라서, OOM과는 거리가 있어보이네요!

**Comment by menthamin** [comment]
MySQL 서버가 메모
리 부족으로 멈춰 버릴 수도 있으므로 적절한 메모리 공간을 설정하는 것이 중요하다. 로컬 메모리 공
간의 또 한 가지 중요한 특징은 각 쿼리의 용도별로 필요할 때만 공간이 할당되고 필요하지 않은 경우
에는 MySQL이 메모리 공간을 할당조차도 하지 않을 수도 있다는 점이다

(책85page)

**Comment by menthamin** [comment]
thread_cache_size
- 기존 5.X 버전에서는 default_size가 0이 였다.
- 8.0 버전이 되면서 default가 9로 할당되도록 변경 되었다.
- 공간 자체가 필요없는 경우에 할당하면 낭비일 수 있다.
